frameworks/PHP/swoole/swoole-server-postgres.php

```diff
@@ -0,0 +1,247 @@
+
+    $pool->put($db);
+
+    return \json_encode($arr, JSON_NUMERIC_CHECK);
+};
+
+
+/**
+ * On every request to the (web)server, execute the following code
+ */
+$server->on('request', function (Request $req, Response $res) use ($db, $fortunes, $updates) {
+    try {
+        switch ($req->server['request_uri']) {
+            case '/json':
+                $res->header('Content-Type', 'application/json');
+                $res->end(json_encode(['message' => 'Hello, World!']));
+                break;
+
+            case '/plaintext':
+                $res->header('Content-Type', 'text/plain; charset=utf-8');
+                $res->end('Hello, World!');
+                break;
+
+            case '/db':
+                $res->header('Content-Type', 'application/json');
+
+                if (isset($req->get['queries'])) {
+                    $res->end($db((int)$req->get['queries']));
+                } else {
+                    $res->end($db(-1));
+                }
+                break;
+
+            case '/fortunes':
+                $res->header('Content-Type', 'text/html; charset=utf-8');
+                $res->end($fortunes());
+                break;
+
+            case '/updates':
+                $res->header('Content-Type', 'application/json');
+
+                if (isset($req->get['queries'])) {
+                    $res->end($updates((int)$req->get['queries']));
+                } else {
+                    $res->end($updates(-1));
+                }
+                break;
+
+            default:
+                $res->status(404);
+                $res->end('Error 404');
+
+        }
+
+    } catch (\Throwable $e) {
+        $res->status(500);
+        $res->end('Error 500');
+    }
+});
+
+$server->start();
+
+/**
+ * Class DatabasePool
+ *
+ * Deal with the fact that Swoole 2.1.3 has no build in database pooling
+ */
+class DatabasePool
+{
+    private $server = [
+        'host' => '',
+        'user' => 'benchmarkdbuser',
+        'password' => 'benchmarkdbpass',
+        'database' => 'hello_world'
+    ];
+
+    private $pool;
+    
+    private $type;
+
+    public function __construct($type)
+    {
+        $this->server['host'] = \gethostbyname('tfb-database');
+        $this->type = $type;
+    }
+    
+    public function init($capacity)
+    {
+        $this->pool=new \Swoole\Coroutine\Channel($capacity);
+        while($capacity>0){
+            $db=$this->createDbInstance();
+            if($db!==false){
+                $this->pool->push($db);
+                $capacity--;
+            }
+        }
+    }
+
+    private function createDbInstance()
+    {
+        if ($this->type === 'postgres') {
+            $db = new Swoole\Coroutine\PostgreSql;
+            if ($db->connect("host={$this->server['host']} port=5432 dbname={$this->server['database']} user={$this->server['user']} password={$this->server['password']}")){
+                return $db;
+            }
+        }
+        
+        return false;
```

Comment:
Why not add in `swoole-server.php` for both mysql and postgres?
So will be only one codebase to maintain.